misc/daemon/minimega.service

```diff
@@ -0,0 +1,30 @@
+[Unit]
+Description="minimega Emulytics Daemon"
+Documentation=https://www.sandia.gov/minimega
+Wants=openvswitch-switch.service
+
+[Install]
+WantedBy=default.target
+
+[Service]
+Type=simple
+# Kill mode settings
+KillMode=control-group
+TimeoutStopSec=5
+Restart=on-success
+# Setting ulimit and environment variables
+LimitNOFILE=1024000
+LimitNPROC=4096000
+EnvironmentFile=/etc/minimega/minimega.conf
+# Pre Start script
+ExecStartPre=/bin/mkdir -p /etc/minimega/saved_vms/
+ExecStartPre=/bin/mkdir -p ${MM_RUN_PATH}
+# Starting minimega
+ExecStart=/usr/bin/minimega -base=${MM_RUN_PATH} -filepath=${MM_FILEPATH} -degree=${MM_MESH_DEGREE} -nostdin=${MM_DAEMON} -context=${MM_CONTEXT} -port=${MM_PORT} -level=${MM_LOG_LEVEL} -logfile=${MM_LOG_FILE} &
```

Comment:
There are a lot of additional variables that have been added to the Dockerized config that aren't present in the `minimega.conf` used by the init daemon (see [start-minimega.sh](https://github.com/sandia-minimega/minimega/blob/master/docker/start-minimega.sh)). Should those be added to this script, as well as `minimega.conf` and the old init script? 

Some examples of variables: `MM_BROADCAST` for `-broadcast`, `MM_VLANRANGE` for `-vlanrange`, and `MM_APPEND` to append additional arguments that don't have variables (this was added recently in #1542 ).

Reply:
Adding the new parameters to the `minimega.conf` and `systemd` file definitely makes sense and I'll push that soon. 

Personally, I don't think it makes sense to update the `init.d` script as that system has largely been removed from most popular Linux distributions and I don't know if it's worth maintaining that code. If fact I think it should probably be deprecated in the next release (and subsequently removed the following release) as Docker and `systemd` are better options.

In looking closer at the various config options it looks like there is a naming mismatch between [start-minimega.sh](https://github.com/sandia-minimega/minimega/blob/master/docker/start-minimega.sh) and [minimega.conf](https://github.com/sandia-minimega/minimega/blob/master/misc/daemon/minimega.conf). For example, `MM_RUN_PATH` is passed in as the `-base` parameter with `minimega.conf` and this variable got renamed to `MM_BASE` in `start-minimega.sh`. I don't think that this PR is necessarily the right place to modify those variable names as 1) It feels like scope creep and 2) I suspect any renaming will cause breaking changes to existing users setup. 

One possible solution is to add "duplicate" variables (e.g, `MM_RUN_PATH` and `MM_BASE` to minimega.conf and move the `systemd` file to use those new ones. Then we can deprecate the "old" variable names and remove them when we remove the `init.d` script in a few versions. Thoughts?

Reply:
I added the new variables (with a systemd preference to use the new version). Let me know if you would like any other changes.

Reply:
The changes look good. I agree with you on all points. 

Regarding the `init.d` scripts, systemd is how most of the major distributions operate nowadays, and there isn't a reason to continue maintaining the old scripts. Better to leave them alone and not risk breaking using still running on older distros.

For duplicate variables, your solution is good, and I don't see any obvious downsides to that approach.

Reply:
Also, just so it's clear this PR will need review by a minimega developer prior to merge. I'm just a humble downstream user with a vested interest in improvements to how it's deployed 😁